snap/patches/nova/0001-remotefs-http-driver.patch

```diff
@@ -0,0 +1,246 @@
++    # Lazy import to keep dependency surface localized
++    from webdav4.client import Client
++
++    base_url = _base_url(host)
++
++    # TLS verification and client certs for mTLS
++    ca_file = getattr(CONF.libvirt, "file_server_ca_file", None) or True
++    cert_file = getattr(CONF.libvirt, "file_server_client_cert_file", "") or None
++    key_file = getattr(CONF.libvirt, "file_server_client_key_file", "") or None
++    client_cert = (cert_file, key_file) if cert_file and key_file else None
++
++    kwargs = {}
++    # Handle SSL verification: verify certificate chain but not hostname
++    # (since we may connect via IP address, not hostname)
++    if ca_file and ca_file is not True:
++        ssl_context = ssl.create_default_context(cafile=ca_file)
++        ssl_context.check_hostname = False
++        kwargs["verify"] = ssl_context
++    elif ca_file is True:
++        ssl_context = ssl.create_default_context()
++        ssl_context.check_hostname = False
++        kwargs["verify"] = ssl_context
++    else:
++        kwargs["verify"] = False
++    
++    if client_cert:
++        kwargs["cert"] = client_cert
++
++    read_timeout = getattr(CONF.libvirt, "file_server_read_timeout", None)
++    if read_timeout:
++        kwargs["timeout"] = (5, int(read_timeout))
++    else:
++        kwargs["timeout"] = (5, 900)
++
++    # Disable proxy usage from environment variables
++    # httpx (used by webdav4) will otherwise try to use HTTP_PROXY/HTTPS_PROXY
++    kwargs["trust_env"] = False
++
++    # If webdav4 requires basic auth in addition to mTLS, uncomment:
++    # kwargs["auth"] = ("nova", "")  # username matches SFTPGo config
++
++    return Client(base_url, **kwargs)
++
++
++def _ensure_remote_dirs(client, path: str) -> None:
++    """Ensure that each component of the remote path exists."""
++    if not path:
++        return
++
++    from webdav4.client import ResourceAlreadyExists
++
++    components = [segment for segment in path.strip("/").split("/") if segment]
++    current = ""
++    for segment in components:
++        current = f"{current}/{segment}" if current else segment
++        current_dir = _ensure_trailing_slash(current)
++        _debug_log_path("ensure_remote_dirs candidate", current_dir)
++        if client.exists(current_dir):
++            continue
++        try:
++            client.mkdir(current_dir)
++            LOG.debug("[webdav] mkdir %s", current_dir)
++        except ResourceAlreadyExists:
++            # Another worker may have created this path between exists() and mkdir()
++            continue
++
++
++class HttpDriver:
++    """HTTP(WebDAV) transport driver implementing RemoteFilesystemDriver API."""
++
++    def create_file(self, host, dst_path, on_execute=None, on_completion=None):
++        LOG.debug("[webdav] create file %s on %s", dst_path, host)
++        client = _new_client(host)
++        webdav_path = dst_path
++
++        parent = os.path.dirname(webdav_path.rstrip("/"))
```

Comment:
What do you think about using pathlib module instead of os.path. We need not deal with string handling if any.